# Descriptives on _Disaster Tweets_


### Import Packages

In [1]:
import pandas as pd
import numpy as np

datapath = 'data/'

### Load Data

In [2]:
data = pd.read_csv(datapath + 'disaster_train.csv', sep=',',)

In [17]:
print('Shape of the data:', data.shape)
data.head()

Shape of the data: (7613, 5)


id keyword location                                               text  \
0   1     NaN      NaN  Our Deeds are the Reason of this #earthquake M...   
1   4     NaN      NaN             Forest fire near La Ronge Sask. Canada   
2   5     NaN      NaN  All residents asked to 'shelter in place' are ...   
3   6     NaN      NaN  13,000 people receive #wildfires evacuation or...   
4   7     NaN      NaN  Just got sent this photo from Ruby #Alaska as ...   

   target  
0       1  
1       1  
2       1  
3       1  
4       1

The dataset contains 7613 tweets 5 features: id, keyword, location, text, and target

In [23]:
print('Missing keywords:', sum(data['keyword'].isna()))
print('Missing locations:', sum(data['location'].isna()))

Missing keywords: 61
Missing locations: 2533


Out of those 7613 61 do not have a keyword an 2533 do not have a location

In [41]:
data_bool = data.copy().loc[:, 'keyword':'target']
data_bool['keyword'] = data_bool['keyword'].isna()
data_bool['location'] = data_bool['location'].isna()
data_bool['count'] = np.ones(data.shape[0], dtype=int)

In [47]:
summ = data_bool.groupby('target').sum()
summ['location_ratio'] = summ['location']/summ['count']
summ

keyword  location  count  location_ratio
target                                          
0            19      1458   4342        0.335790
1            42      1075   3271        0.328646

Grouping by target, there are 4342 fake tweets (target: 0) and 3271 real tweets (target: 1). The ratio of missing locations is similar, the amount of missing keyqwords is negligibly different. 
